contrib/hyperscan/matching/input_matchers/source/matcher.cc

```diff
@@ -152,6 +155,21 @@ void Matcher::compile(const std::vector<const char*>& expressions,
   hs_free_compile_error(compile_err);
 }
 
+hs_scratch_t* Matcher::getScratch(ScratchThreadLocalPtr& local_scratch) const {
```

Comment:
can you add comments here describing the init-order challenges you are overcoming with this scratch buffer?

Reply:
Ok, I have added some comments in the methods.